projects/canopy/src/lib/forms/checkbox-group/checkbox-group.component.ts

```diff
@@ -0,0 +1,156 @@
+import {
+  Component,
+  ContentChild,
+  ContentChildren,
+  forwardRef,
+  Host,
+  HostBinding,
+  Input,
+  Optional,
+  QueryList,
+  Self,
+  SkipSelf,
+  ViewEncapsulation,
+} from '@angular/core';
+import { ControlValueAccessor, FormGroupDirective, NgControl } from '@angular/forms';
+
+import { LgDomService } from '../../utils/dom.service';
+import { LgHintComponent } from '../hint/hint.component';
+import { LgErrorStateMatcher } from '../validation/error-state-matcher';
+import { LgValidationComponent } from '../validation/validation.component';
+import { LgToggleComponent } from '../toggle';
+
+let nextUniqueId = 0;
+
+@Component({
+  selector: 'lg-checkbox-group',
+  templateUrl: './checkbox-group.component.html',
+  styleUrls: ['./checkbox-group.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+})
+export class LgCheckboxGroupComponent implements ControlValueAccessor {
+  private _name = `lg-checkbox-group-${nextUniqueId++}`;
+
+  @Input() id = `lg-checkbox-group-id-${nextUniqueId++}`;
```

Comment:
Would this increment `nextUniqueId ` twice, resulting in two different IDs?

Reply:
Yep, will fix this.

Reply:
Done.

Reply:
Sorry to add another request, but maybe we need a unit test to cover this?

Reply:
No worries.
I've done this for both checkbox group and radio group.
Radio group fix in a separate commit.